### ISSUE TITLE
[docs][bug] multiple elements with `id="summary"` on subpages

### ISSUE THREAD

**Issue by hellogreg** [RedHat-UX/red-hat-design-system#1849]
Across ux dot subpages---230+ times, according to our Lumar crawler--the "On this page" text is within two elements per page: one for smaller screens (within a `<summary>`) and one for larger screens (within a `<div>`).

Both elements have the same `id` value: `"summary"`.

Though one of these elements should always have a `display: none` applied to it (assuming CSS is enabled), this still can cause validation and potential assistive tech issues--especially as there is an `aria-describedby` pointing to this `id`. 

I'd recommend giving these elements different `id` values (and then reworking site JS and CSS to account for this).

![Screencap of a page inspector showing the duplicate id values](https://github.com/user-attachments/assets/6cf0c9cc-1645-4c80-a7a4-550aee046210)

**Comment by zeroedin** [comment]
This is a side effect from DSD'ing the component.  Previously only 1 of these blocks would render to the screen at a time.

**Comment by bennypowers** [comment]
Is this a problem on the clefairy branch?

**Comment by zeroedin** [comment]
> Is this a problem on the clefairy branch?

Looks like this is resolved in the clefairy branch.  

Page Tested with axe dev tools and site improve https://deploy-preview-1802--red-hat-design-system.netlify.app/foundations/color/ 

Greg mind verifying? 

Other issue's we'll want to clean up found.

**Comment by hellogreg** [comment]
Thanks, @zeroedin ! So, the good news is that this does work with _JS_ disabled. But with _CSS_ disabled, there still are two items with the same `id`:

![Screencap of CSS-disabled page showing elements with duplicate ids](https://github.com/user-attachments/assets/e86a0f56-4e49-4596-91ed-a80f0ab5e089)

While 1) most modern assistive tech does account for CSS, and 2) both of those elements have the same text content (so an `aria-describedby` will always return "On this page"). _Some_ assistive tech may ignore the CSS, and it still will trigger errors in some validators/tools (like Lumar).

Is that `id` attribute critical? Can it be a class? And/or can it not have different values for the two usages of it?

Actually, now that I think about it... Are we _only_ using that `id` because of the `aria-describedby` reference from the `<nav>` in the `<details>`? (And why only on the mobile `<nav>` and not the desktop?)

If that is the case, could we do this?
- First, use `aria-label` or `aria-labelledby` on both versions of the `<nav>`. One of those seems more appropriate than `aria-describedby` imo.
- If we use `aria-labelledby`, we'd still have the same `id` duplication/reference issue.
- But... if we just use `aria-label` to duplicate the "On this page" text, we can get rid of the `id` altogether.

In other words, the resultant HTML would be like this:
```
<!-- small screen -->
<details>
  <summary>On this page</summary>
  <nav aria-label="On this page">...</nav>
</details>
<!-- big screen -->
<div id="expanded">
  <div>On this page</div>
  <nav aria-label="On this page">...</nav>
</div>
```

**Comment by zeroedin** [comment]
> * If we use `aria-labelledby`, we'd still have the same `id` duplication/reference issue.

Shouldn't be an issue:

Here is what the render method for `uxdot-toc` would look like given that change.
```
 render() {
    return html`
      <aside id="container" part="container">
        <details>
          <summary>${this.summary}</summary>
          <nav aria-label="${this.summary}">
            <slot name="details"></slot>
          </nav>
        </details>
        <div id="expanded">
          <div>${this.summary}</div>
          <nav aria-label="${this.summary}">
            <slot name="expanded"></slot>
          </nav>
        </div>
      </aside>
    `;
  }
}
```

**Comment by hellogreg** [comment]
Nice! Exactly the kind of thing I was hoping would be possible.

**Comment by bennypowers** [comment]
I believe we fixed this yesterday

**Comment by adamjohnson** [comment]
Just following up here too: looks like this issue still persisted in `clefairy`, so I pushed up #1898 to fix it. With Greg's approval on that PR, we can close out this issue. 

Greg: follow up here after another Lumar crawl to make sure it's cleared from there as well. TY!

**Comment by adamjohnson** [comment]
Closed by #1898.
